xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -1022,7 +1022,7 @@ void CPVRClients::StartChannelScan(void)
   long perfCnt = XbmcThreads::SystemClockMillis();
 
   /* stop the supervisor thread */
-  g_PVRManager.StopUpdateThreads();
+  g_PVRManager.Stop();
```

Comment:
don't have time to check what you've turned the pvrmanager into and obviously didn't follow recent dev, but unless `CPVRManager::Stop()` no longer deletes it's data, this is going to crash and burn when you touch the add-on 2 lines down.

but if  `CPVRManager::Stop()` no longer deletes all it's data, then restarting the manager will no longer reload it's data from scratch? and if it does still delete it's data but uses smart pointer magic to keep the `scanClient` instance alive, then keep this in mind: loading the pvr manager and all it's data is _very_ expensive and slow on small arm stuff.